README.md
# Basejump WordPress theme

## Introduction

Some people swear by frameworks, some of us work best with grids and yet others prefer to build child themes off the WordPress default theme(s).

Having worked with all of the above, they have one thing in common: there's always something.

* A framework tends to try to cover all its bases and is therefore by default adding way too much "fluff" to the equation.
* When developing based on a grid - whether it's 12 columns, 16 columns, the golden ratio or whatever - there comes a point that a design is just that bit different and you will need to introduce all kinds of div's and/or classes to make the design fit the grid.
* Child themes are nice most of the times, but you need to rewrite a lot of functions or you're stuck with code that never gets used; in other words you'll end up with unnecessary overhead. 

That has been the main reason for me to develop this Basejump WordPress theme.

With the name I have tried to already suggest what it is. Take this theme and "jump off a cliff" (not literally of course) to make it into the site you want it to become.

## Features

Although the Basejump WordPress theme is a base theme for you to build upon, it still comes packed with quite a few features and I have coded it for WordPress 3.4 and above.

* Six different Page Templates, neatly organized in their own Page Templates directory (WordPress 3.4 feature)
* Upto six columns (best used on the Full Width Page Template)
* Superfish navigation dropdown menu
* Fully internationalized
* WPML ready and when activated it shows the language menu right in the menu bar navigation
* Custom background with default background when no custom background has been uploaded by the user; coded with add_theme_support (WordPress 3.4 feature)
* Seven "sidebars" including 4 in the footer that can be used for widgets
* 4 Social Media share buttons in post footer on single post template
* Responsive

## Installation

The name of the downloaded zip file from Github looks something like senlin-basejump-####.zip and the extracted directory will have the same name. I'd suggest to rename the directory to "basejump" or give it any other name you wish. If you do the latter, make sure to have your stylesheet reflect that name too.

Upload the entire basejump directory to the wp-content folder and activate it through the WordPress Dashboard.

After activation the first thing to do is to add a navigation menu and appoint that menu to be the Primary. Second I suggest to go to the widgets and delete all the widgets that WordPress automatically adds to the first available widget on a fresh install.

Only if you do these two things first, will all the features of the theme work out of the box and will the theme immediately look good.

## FAQ
### Why is the Basejump WordPress Theme not downloadable via the WordPress Theme Repository?

Good question!
Not because I am lazy. The main reason is that I have built the Basejump WordPress Theme with future development (for clients) in mind. Not sure about you, but none of my clients ever has wanted to be able to change the header of the theme. My clients actually want to show their company-logo in the header. One of the requirements of listing a theme in the official WordPress Theme Repository is that you need to have the `add_theme_support('custom-header')` functionality included. As I don't do custom headers the Basejump WordPress Theme therefore cannot be included in the Repository.

### Why does Basejump look so much like the default Genesis child theme?

Fans of Genesis will indeed quickly see that Basejump looks very much like the default Child theme of that premium framework. I actually have tried to work with Genesis, but gave up because to me it is just not a natural way of putting together a website. The reason that my Basejump theme looks so much alike is because I very much like the structure of how the Genesis default child theme has been set up. So I decided to rip it apart and use the structure to base my theme on.

### How to code the columns as can be seen on the Page "up to 6 columns"?

On the Page [up to 6 columns](http://wpbaseju.mp/page-templates/full-width/upto-6-columns/) you can see that you can add up to 6 columns. Adding these columns is done with classes as can be seen in the source code of that page. For your convenience I have made a [Gist](https://gist.github.com/3011643), so you can also easily copy it from there. Probably overkill, but nevertheless: although it works on all page templates, more than 4 columns is best used on the full width page template.

## TODO LIST

<del>To make a default .pot and .po file</del>

## Comments / Questions

You can leave any comments and/or ask any questions in the forum on the [Basejump website](http://wpbaseju.mp/forums/)

## Demo

Since the end of June 2012 The Basejump WordPress Theme has it's own demo domain at [http://wpbaseju.mp](http://wpbaseju.mp). You can see another live site where I have used the theme [here](http://senlinhostingclub.com).

## Download (or Fork)
The Basejump WordPress Theme can be downloaded (or forked) from [GitHub](https://github.com/senlin/basejump).

## Disclaimer

The Basejump WordPress theme runs on WordPress version 3.4 and above and I offer no backward compatibility.

change_log.txt
------------------------------------------------------------------------------------------------------------------
Version 0.2.4
	- added the Dutch language file to the languages directory (languages/basejump-nl_NL.mo)

------------------------------------------------------------------------------------------------------------------
Version 0.2.3
	- adjusted the enqueue scripts function to be more general
	- added languages directory with basejump.pot and basejump.po files

------------------------------------------------------------------------------------------------------------------
Version 0.2.2
	- discovered that Version 0.2.0 made footer widgets not responsive, so corrected that

------------------------------------------------------------------------------------------------------------------
Version 0.2.1
	- renamed functions to be more consequent throughout the theme

------------------------------------------------------------------------------------------------------------------
Version 0.2.0
	- Changed something in comments structure for Pages
	- Changed styling top navigation menu
	- Changed default footer-widgets coding a little bit around
	- Changed location of <?php wp_footer(); ?>
	- Changed styling various native WP elements
	- Changed post header in loop (general and category)
	- Deleted empty duplicate of Edit link

------------------------------------------------------------------------------------------------------------------
Version 0.1.1
	- Added intro text to readme.md
	- Added screenshot.png

------------------------------------------------------------------------------------------------------------------
Version 0.1.0
	- Basejump WordPress theme ready for use

footer.php
	<?php get_template_part( 'footer', 'widgets' ); ?>
	<div id="footer" class="footer">
		<div class="wrap">
			<div class="gototop">
				<p><a rel="nofollow" href="#wrap"><?php _e('Go to top ', 'basejump'); ?>&uarr;</a></p>
			</div><!-- end .gototop -->
			<div class="creds">
				<p><?php _e('&copy; Copyright ', 'basejump'); ?><?php echo date('Y'); ?> &middot; <a href="<?php echo home_url(); ?>" title="<?php bloginfo('name'); ?>"><?php bloginfo('name'); ?></a> &middot; <a href="http://senl.in/baseJumP" title="<?php _e('Basejump WordPress Theme ', 'basejump'); ?>"><?php _e('Basejump WordPress Theme ', 'basejump'); ?></a><?php _e('developed by ', 'basejump'); ?><a href="http://senlinonline.com" title="<?php _e('Basejump WordPress Theme developed by Senlin Online', 'basejump'); ?>">Senlin Online</a> &middot; <?php _e('Powered by ', 'basejump'); ?><a href="http://wordpress.org" target="_blank" title="<?php _e('Powered by WordPress', 'basejump'); ?>"><?php _e('WordPress', 'basejump'); ?></a></p>	
			</div><!-- end .creds -->
		</div><!-- end .wrap -->
	</div><!-- end #footer -->
</div><!-- end #wrap -->
<?php wp_footer(); ?>
<script>!function(d,s,id){var js,fjs=d.getElementsByTagName(s)[0];if(!d.getElementById(id)){js=d.createElement(s);js.id=id;js.src="//platform.twitter.com/widgets.js";fjs.parentNode.insertBefore(js,fjs);}}(document,"script","twitter-wjs");</script>
<script type="text/javascript">
  (function() {
    var po = document.createElement('script'); po.type = 'text/javascript'; po.async = true;
    po.src = 'https://apis.google.com/js/plusone.js';
    var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(po, s);
  })();
</script>
</body>
</html>

header.php
<!DOCTYPE html>
<html <?php language_attributes(); ?>>
<head>
<meta charset="<?php bloginfo( 'charset' ); ?>" />
<title><?php global $page, $paged; wp_title( '|', true, 'right' );
// Add the blog name.
bloginfo( 'name' );
// Add the blog description for the home/front page.
$site_description = get_bloginfo( 'description', 'display' );
if ( $site_description && ( is_home() || is_front_page() ) ) echo " | $site_description"; ?></title>
<link rel="profile" href="http://gmpg.org/xfn/11" />
<meta name="viewport" content="width=device-width, initial-scale=1.0"/>
<!--[if lt IE 9]>
	<script src="http://css3-mediaqueries-js.googlecode.com/svn/trunk/css3-mediaqueries.js"></script>
<![endif]-->
<link rel="stylesheet" type="text/css" media="all" href="<?php bloginfo( 'stylesheet_url' ); ?>" />
<!-- /* @license MyFonts Webfont Build ID 2229308, 2012-04-13T00:49:51-0400 The fonts listed in this notice are subject to the End User License Agreement(s) entered into by the website owner. All other parties are explicitly restricted from using the Licensed Webfonts(s). You may obtain a valid license at the URLs below. Webfont: Flexo Caps DEMO by Durotype URL: http://www.myfonts.com/fonts/durotype/flexo/demo/ Copyright: Copyright &#x00A9; 2011-2012 Durotype, www.durotype.com. All rights reserved. Licensed pageviews: unlimited License: http://www.myfonts.com/viewlicense?type=web&buildid=2229308 © 2012 Bitstream Inc */ -->
<link rel="stylesheet" type="text/css" href="<?php echo get_stylesheet_directory_uri(); ?>/fonts/flexo.css">
<link rel="pingback" href="<?php bloginfo( 'pingback_url' ); ?>" />
<?php if ( is_singular() && get_option( 'thread_comments' ) )
	wp_enqueue_script( 'comment-reply' );
	wp_head(); ?>
<link rel="shortcut icon" href="<?php echo get_stylesheet_directory_uri(); ?>/images/favicon16.ico"/>
</head>
<body <?php body_class(); ?>>
<div id="wrap">
	<div id="header">
		<div class="wrap">
			<div id="title-area">
				<?php $heading_tag = ( is_home() || is_front_page() ) ? 'h1' : 'div'; ?>
				<<?php echo $heading_tag; ?> id="title">
					<span>
						<a href="<?php echo home_url( '/' ); ?>" title="<?php echo esc_attr( get_bloginfo( 'name', 'display' ) ); ?>" rel="home"><?php bloginfo( 'name' ); ?></a>
					</span>
				</<?php echo $heading_tag; ?>>
				<div id="description"><?php bloginfo( 'description' ); ?></div>
			</div><!-- end #title-area -->
			<?php if ( !function_exists('dynamic_sidebar') || !dynamic_sidebar("Header Widget") ) : ?><?php endif; ?>
		</div><!-- end .wrap -->
	</div><!--end #header-->

	<?php if ( is_404() ) {
	} else { ?>
		<div id="main-image"><img src="<?php echo get_stylesheet_directory_uri(); ?>/images/basejump.jpg" alt="main image" title="Basejump Theme" /></div>
	<?php } ?>
	
	<?php // Check to see whether WPML is active, if it is we include the Language Switcher in the navigation menu and internationalize the date field on the right - source: http://webdesign.onyou.ch/2010/07/28/check-if-a-wordpress-plugin-is-active/ ?>
	<?php if (in_array( 'sitepress-multilingual-cms/sitepress.php', apply_filters( 'active_plugins', get_option( 'active_plugins' ) ) )) {
		wp_nav_menu( array('container_id' => 'nav', 'items_wrap' => '<ul id="%1$s" class="menu menu-primary superfish sf-js-enabled">%3$s' .wpml_language_switch().'<li class="right date">'.date_i18n(get_option('date_format') ,strtotime("11/15-1976")).'</li></ul>', 'theme_location' => 'primary' ) );
	} else {
		wp_nav_menu( array('container_id' => 'nav', 'items_wrap' => '<ul id="%1$s" class="menu menu-primary superfish sf-js-enabled">%3$s <li class="right date">'.date_i18n(get_option('date_format')).'</li></ul>', 'theme_location' => 'primary' ) );
	} ?>
	
	<div id="inner">
	<?php if ( ! is_home() && ! is_404()) {
		if (function_exists('so_write_breadcrumb') ) { 
			echo '<div class="breadcrumb">';
			echo so_write_breadcrumb();
			echo '</div><!-- .breadcrumb -->';
		}
	}
	?>

loop-single.php
				<div <?php post_class(); ?>>
					<h1 class="entry-title"><?php the_title(); ?></h1>
					<div class="post-info">
						<?php so_posted_on(); ?> <span class="post-comments"><?php /* http://www.social-ink.net/blog/wordpress-tips-removing-comments-off */ comments_popup_link( '<span class="leave-reply"></span>', _x( '1 Comment', 'comments number', 'basejump' ), _x( '% Comments', 'comments number', 'basejump' ), '', '' ); ?></span> <?php edit_post_link( __( '(Edit)', 'basejump' ) ); ?>
					</div><!-- end .postinfo -->
					<div class="entry-content">
						<?php 
						 if ( has_post_thumbnail()) {
						   $large_image_url = wp_get_attachment_image_src( get_post_thumbnail_id(), 'large');
						   echo '<div class="feat-img"><a href="' . $large_image_url[0] . '" title="' . the_title_attribute('echo=0') . '" >';
						   the_post_thumbnail('single');
						   echo '</a></div><!-- end .feat-img -->';
						 }
						 ?>
						<?php the_content(); ?>
						<?php wp_link_pages( array( 'before' => '<div class="page-link">' . __( 'Pages:', 'basejump' ), 'after' => '</div>' ) ); ?>
					</div><!-- end .entry-content -->
					<div class="post-meta">
						<?php if ( count( get_the_category() ) ) : ?>
							<span class="categories">
								<?php printf( __( '<span class="%1$s">Categorized under</span> %2$s ', 'basejump' ), 'entry-utility-prep entry-utility-prep-cat-links', get_the_category_list( ', ' ) ); ?>
							</span>
							<span class="tags">
								<?php
								$tag = get_the_tags();
								if (! $tag) { 
									echo __( 'and untagged', 'basejump' );
								} else {
									echo __( 'and tagged with ', 'basejump' );
									the_tags( '', ', ' );
								} ?>
							</span>
						<?php endif; ?>
						<?php get_template_part( 'icons', 'social' ); ?>
					</div><!-- end .post-meta -->
				</div><!-- end .post_class -->

loop-general.php
				<div <?php post_class(); ?>>
					<h2 class="entry-title">
						<a href="<?php the_permalink(); ?>" title="<?php the_title_attribute(); ?>" >
							<?php the_title(); ?>
						</a>
					</h2>
					<div class="post-info">
						<?php so_posted_on(); ?> <span class="post-comments"><?php /* http://www.social-ink.net/blog/wordpress-tips-removing-comments-off */ comments_popup_link( '<span class="leave-reply"></span>', _x( '1 Comment', 'comments number', 'basejump' ), _x( '% Comments', 'comments number', 'basejump' ), '', '' ); ?></span> <?php edit_post_link( __( '(Edit)', 'basejump' ) ); ?>
					</div><!-- end .postinfo -->
					<div class="entry-content">
						<?php if ( has_post_thumbnail()) : ?>
							<a href="<?php the_permalink(); ?>" title="<?php the_title_attribute(); ?>" >
								<?php the_post_thumbnail('thumbnail', array('class' => 'alignleft post-image')); ?>
							</a>
						<?php endif; ?>
						<?php the_excerpt(); ?>
					</div><!-- end .entry-content -->
					<div class="post-meta">
						<?php if ( count( get_the_category() ) ) : ?>
							<span class="categories">
								<?php printf( __( '<span class="%1$s">Categorized under</span> %2$s', 'basejump' ), 'entry-utility-prep entry-utility-prep-cat-links', get_the_category_list( ', ' ) ); ?>
							</span>
						<?php endif; ?>
					</div><!-- end .post-meta -->
				</div><!-- end .post_class -->

loop-category.php
				<div <?php post_class(); ?>>
					<h2 class="entry-title">
						<a href="<?php the_permalink(); ?>" title="<?php the_title_attribute(); ?>" >
							<?php the_title(); ?>
						</a>
					</h2>
					<div class="post-info">
						<?php so_posted_on(); ?> <span class="post-comments"><?php /* http://www.social-ink.net/blog/wordpress-tips-removing-comments-off */ comments_popup_link( '<span class="leave-reply"></span>', _x( '1 Comment', 'comments number', 'basejump' ), _x( '% Comments', 'comments number', 'basejump' ), '', '' ); ?></span> <?php edit_post_link( __( '(Edit)', 'basejump' ) ); ?>
					</div><!-- end .postinfo -->
					<div class="entry-content">
						<?php if ( has_post_thumbnail()) : ?>
							<a href="<?php the_permalink(); ?>" title="<?php the_title_attribute(); ?>" >
								<?php the_post_thumbnail('thumbnail', array('class' => 'alignleft post-image')); ?>
							</a>
						<?php endif; ?>
						<?php the_excerpt(); ?>
					</div><!-- end .entry-content -->
				</div><!-- end .post_class -->

footer-widgets.php
<div id="footer-widgets">
    <div class="footer-widgets-1">
        <?php if ( !function_exists('dynamic_sidebar') || !dynamic_sidebar('Footer #1') ) : ?>
            <h4 class="widgettitle"><?php _e("Footer Widget 1", 'basejump'); ?></h4>
            <p><?php _e("This is an example of a widgeted area that you can place text to describe a particular product or service. You can also use other WordPress widgets such as recent posts, recent comments, a tag cloud or more.", 'basejump'); ?></p>
	    <?php endif; ?> 
    </div><!-- end .footer-widgets-1 -->
    <div class="footer-widgets-2">
        <?php if ( !function_exists('dynamic_sidebar') || !dynamic_sidebar('Footer #2') ) : ?>
            <h4 class="widgettitle"><?php _e("Footer Widget 2", 'basejump'); ?></h4>
            <p><?php _e("This is an example of a widgeted area that you can place text to describe a particular product or service. You can also use other WordPress widgets such as recent posts, recent comments, a tag cloud or more.", 'basejump'); ?></p>
	    <?php endif; ?> 
    </div><!-- end .footer-widgets-2 -->
    <div class="footer-widgets-3">
        <?php if ( !function_exists('dynamic_sidebar') || !dynamic_sidebar('Footer #3') ) : ?>
            <h4 class="widgettitle"><?php _e("Footer Widget 3", 'basejump'); ?></h4>
            <p><?php _e("This is an example of a widgeted area that you can place text to describe a particular product or service. You can also use other WordPress widgets such as recent posts, recent comments, a tag cloud or more.", 'basejump'); ?></p>
	    <?php endif; ?> 
    </div><!-- end .footer-widgets-3 -->
	    <div class="footer-widgets-4">
	        <?php if ( !function_exists('dynamic_sidebar') || !dynamic_sidebar('Footer #4') ) : ?>
	            <h4 class="widgettitle"><?php _e("Footer Widget 4", 'basejump'); ?></h4>
	            <p><?php _e("This is an example of a widgeted area that you can place text to describe a particular product or service. You can also use other WordPress widgets such as recent posts, recent comments, a tag cloud or more.", 'basejump'); ?></p>
		    <?php endif; ?> 
	    </div><!-- end .footer-widgets-4 -->
</div><!-- end #footer-widgets -->

functions.php
<?php
/* Basejump Functions File */

// Include Functionality & Dashboard Functions Files
//require_once 'lib/dashboard-functions.php';

// Set the content width based on the theme's design and stylesheet; copied from Twentyeleven
if ( ! isset( $content_width ) )
	$content_width = 640;


/* Do theme setup on the 'after_setup_theme' hook. */
add_action( 'after_setup_theme', 'basejump_theme_setup' );
function basejump_theme_setup() {
	
	// Add theme support for various things
	add_theme_support( 'automatic-feed-links' );
	add_theme_support( 'post-thumbnails' );
	add_theme_support( 'menus' );
	// Add functionality to change Custom Background from WP 3.4 onwards this has been changed to add_theme_support( 'custom-background', $args )
	// sources: http://kovshenin.com/2012/default-custom-background-for-wordpress-3-4/ & http://make.wordpress.org/themes/2012/04/06/updating-custom-backgrounds-and-custom-headers-for-wordpress-3-4/
	add_theme_support( 'custom-background', array(
		'default-image' => get_stylesheet_directory_uri() . '/images/first_aid_kit.png' 
	) );
	// uncomment the line below to allow post-formats
	//add_theme_support( 'post-formats', array( 'aside', 'gallery', 'link', 'image', 'quote', 'status', 'video', 'audio', 'chat' ) );
	
	// Additional Image Sizes
	add_image_size ('single', 640, 9999);

	// Make theme available for translation
	// Translations can be filed in the /languages/ directory
	load_theme_textdomain( 'basejump', get_template_directory() . '/languages' );

	$locale = get_locale();
	$locale_file = get_template_directory() . "/languages/$locale.php";
	if ( is_readable( $locale_file ) )
		require_once( $locale_file );

	// This theme uses wp_nav_menu() in one location.
	register_nav_menus( array(
		'primary' => __( 'Primary Navigation', 'basejump' ),
	) );
	
	// SIDEBARS
	function basejump_widgets_init() {
		register_sidebar(array(
			'name' => __('Header Widget', 'basejump'),
			'id' => 'header-promo',
			'description' => __('Optional Widget in Header', 'basejump'),
			'before_widget' => '<div class="widget-area"><div id="%1$s" class="widget %2$s">',
			'after_widget' => '</div></div><!-- end .widget-area -->',
			'before_title'=>'<h4 class="widgettitle">','after_title'=>'</h4>'
		));
	    register_sidebar(array(
			'name' => __('Narrow Sidebar', 'basejump'),
			'description' => __('Narrow Sidebar', 'basejump'),
			'id' => 'sidebar-alt',
			'before_widget' => '<div class="widget-area"><div id="%1$s" class="widget %2$s">',
			'after_widget' => '</div></div><!-- end .widget-area -->',
			'before_title'=>'<h4 class="widgettitle">','after_title'=>'</h4>'
		));
		register_sidebar(array(
			'name' => __('Wide Sidebar', 'basejump'),
			'description' => __('Wide Sidebar', 'basejump'),
			'id' => 'sidebar',
			'before_widget' => '<div class="widget-area"><div id="%1$s" class="widget %2$s">',
			'after_widget' => '</div></div><!-- end .widget-area -->',
			'before_title'=>'<h4 class="widgettitle">','after_title'=>'</h4>'
		));
		register_sidebar(array(
			'name'=> __('Footer #1', 'basejump'),
			'id' => 'footer-1',
			'description' => __('This is the first column of the footer section.', 'basejump'),
			'before_widget' => '<div id="%1$s" class="widget %2$s">',
			'after_widget' => '</div><!-- end .widget -->',
			'before_title'=>'<h4 class="widgettitle">','after_title'=>'</h4>'
		));
		register_sidebar(array(
			'name'=> __('Footer #2', 'basejump'),
			'id' => 'footer-2',
			'description' => __('This is the second column of the footer section.', 'basejump'),
			'before_widget' => '<div id="%1$s" class="widget %2$s">',
			'after_widget' => '</div><!-- end .widget -->',
			'before_title'=>'<h4 class="widgettitle">','after_title'=>'</h4>'
		));
		register_sidebar(array(
			'name'=> __('Footer #3', 'basejump'),
			'id' => 'footer-3',
			'description' => __('This is the third column of the footer section.', 'basejump'),
			'before_widget' => '<div id="%1$s" class="widget %2$s">',
			'after_widget' => '</div><!-- end .widget -->',
			'before_title'=>'<h4 class="widgettitle">','after_title'=>'</h4>'
		));
		register_sidebar(array(
			'name'=> __('Footer #4', 'basejump'),
			'id' => 'footer-4',
			'description' => __('This is the fourth column of the footer section.', 'basejump'),
			'before_widget' => '<div id="%1$s" class="widget %2$s">',
			'after_widget' => '</div><!-- end .widget -->',
			'before_title'=>'<h4 class="widgettitle">','after_title'=>'</h4>'
		));

	}
	/** Register sidebars by running basejump_widgets_init() on the widgets_init hook. */
	add_action( 'widgets_init', 'basejump_widgets_init' );

}

// enqueue Superfish scripts
function basejump_superfish_scripts() {
	wp_enqueue_script( 'jquery' );
	wp_enqueue_script( 'superfish', get_template_directory_uri() . '/js/superfish.js?ver=1.4.8', 'jquery', true );
	wp_enqueue_script( 'superfish-args', get_template_directory_uri() . '/js/superfish.args.js?ver=1.8.0', 'jquery', true );
}
add_action('init', 'basejump_superfish_scripts');

// Add breadcrumb
function basejump_write_breadcrumb() {
    $pid = $post->ID;
	$trail = '<a href="' .home_url( '/' ). '">' . __('Home', 'basejump'). '</a>';
 
    if ( is_front_page() ) {
        // do nothing
	}
		elseif (is_page()) {
			$bcarray = array();
			$pdata = get_post($pid);
			$bcarray[] = ' &raquo; '.$pdata->post_title."\n";
			while ($pdata->post_parent) {
				$pdata = get_post($pdata->post_parent);
				$bcarray[] = ' &raquo; <a href="'.get_permalink($pdata->ID).'">'.$pdata->post_title.'</a>';
			}
		   $bcarray = array_reverse($bcarray);
			 foreach ($bcarray AS $listitem) {
				 $trail .= $listitem;
			 }
		}
		elseif (is_single()) {
			$pdata = get_the_category($pid);
			$adata = get_post($pid);
			if(!empty($pdata)){
				$data = get_category_parents($pdata[0]->cat_ID, TRUE, ' &raquo; ');
				$trail .= " &raquo; ".substr($data,0,-8);
			}
			$trail.= ' &raquo; '.$adata->post_title."\n";
		}
	   	elseif (is_category()) {
			$pdata = get_the_category($pid);
			$data = get_category_parents($pdata[0]->cat_ID, TRUE, ' &raquo; ');
			if(!empty($pdata)){
				$trail .= " &raquo; ".substr($data,0,-8);
			}
	   	}
 $trail.="";
 return $trail;
}

function basejump_posted_on() {
	printf( __( '<span class="%1$s">Posted on</span> %2$s', 'basejump' ),
		'meta-prep meta-prep-author',
		sprintf( '<span class="entry-date">%1$s</span>',
			get_the_date()
		)
	);
}

/* Make date format (for use in nav bar) properly translatable 
source: http://wpml.org/forums/topic/date-format-isnt-translated-is-that-possible-done-by-wpml/#post-30197
*/
function basejump_update_date_format($newvalue, $oldvalue) {
	if(function_exists('icl_register_string')) {
		icl_register_string('WP', 'date_format', $newvalue);
	}
	return $newvalue;
}
add_filter( 'pre_update_option_date_format', 'basejump_update_date_format', 10, 2);

function basejump_date_format($value) {
	if(function_exists('icl_t')) {
		return icl_t('WP', 'date_format', $value);
	}
	return $value;
}
add_filter( 'option_date_format', 'basejump_date_format');

/* Custom Language Switch automatically added to navbar
Source: http://wpml.org/2009/10/building-a-flexible-multilingual-content-site-with-wordpress/ Scroll down to Language Switching
*/
function wpml_language_switch() {
	$lang = icl_get_languages('skip_missing=N');
	$ret = '';
	if(count($lang) > 0) {
	foreach($lang as $value) {
		$ret .= ($value['active'] == 0) ? '<li class="language dropdown menu-item"><a href="' . $value['url'] . '">' .
			$value['native_name']  . '</a></li>' : '';
	}
	}
	return $ret;
}

// Comments template
if ( ! function_exists( 'basejump_comment' ) ) :
/**
 * Template for comments and pingbacks.
 *
 * To override this walker in a child theme without modifying the comments template
 * simply create your own basejump_comment(), and that function will be used instead.
 *
 * Used as a callback by wp_list_comments() for displaying the comments.
 *
 * Copied from Twenty Eleven
 */
function basejump_comment( $comment, $args, $depth ) {
	$GLOBALS['comment'] = $comment;
	switch ( $comment->comment_type ) :
		case 'pingback' :
		case 'trackback' :
	?>
	<li class="post pingback">
		<p><?php _e( 'Pingback:', 'basejump' ); ?> <?php comment_author_link(); ?><?php edit_comment_link( __( 'Edit', 'basejump' ), '<span class="edit-link">', '</span>' ); ?></p>
	<?php
			break;
		default :
	?>
	<li <?php comment_class(); ?> id="li-comment-<?php comment_ID(); ?>">
		<div id="comment id-<?php comment_ID(); ?>" class="comment">
			<div class="comment-author vcard">
				<?php
					$avatar_size = 48;
					if ( '0' != $comment->comment_parent )
						$avatar_size = 48;

					echo get_avatar( $comment, $avatar_size );

					/* translators: 1: comment author, 2: date and time */
					printf( __( '%1$s on %2$s <span class="says">said:</span>', 'basejump' ),
						sprintf( '<cite class="fn">%s</cite>', get_comment_author_link() ),
						sprintf( '<a href="%1$s"><time pubdate datetime="%2$s">%3$s</time></a>',
							esc_url( get_comment_link( $comment->comment_ID ) ),
							get_comment_time( 'c' ),
							/* translators: 1: date, 2: time */
							sprintf( __( '%1$s at %2$s', 'basejump' ), get_comment_date(), get_comment_time() )
						)
					);
				?>

				<?php edit_comment_link( __( 'Edit', 'basejump' ), '<span class="edit-link">', '</span>' ); ?>
			</div><!-- .comment-author .vcard -->
			<div class="comment-meta">
				

				<?php if ( $comment->comment_approved == '0' ) : ?>
					<em class="comment-awaiting-moderation"><?php _e( 'Your comment is awaiting moderation.', 'basejump' ); ?></em>
					<br />
				<?php endif; ?>

			</div>

			<div class="comment-content"><?php comment_text(); ?></div>

			<div class="reply">
				<?php comment_reply_link( array_merge( $args, array( 'reply_text' => __( 'Reply <span>&darr;</span>', 'basejump' ), 'depth' => $depth, 'max_depth' => $args['max_depth'] ) ) ); ?>
			</div><!-- .reply -->
		</div><!-- #comment-## -->

	<?php
			break;
	endswitch;
}
endif; // ends check for basejump_comment()

/**
 * Adds two classes to the array of body classes.
 * The first is if the site has only had one author with published posts.
 * The second is if a singular post being displayed
 *
 * Copied from Twenty Eleven
 */
function basejump_body_classes( $classes ) {

	if ( function_exists( 'is_multi_author' ) && ! is_multi_author() )
		$classes[] = 'single-author';

	if ( is_singular() && ! is_home() && ! is_page_template( 'showcase.php' ) && ! is_page_template( 'sidebar-page.php' ) )
		$classes[] = 'singular';

	return $classes;
}
add_filter( 'body_class', 'basejump_body_classes' );

// Define the comment_form() function here
add_action( 'comment_form', 'basejump_comment_form' );
/**
 * Defines the comment form, hooked to <code>comment_form()</code>
 *
 * copied from Genesis
 */
function basejump_comment_form() {

	/** Bail if comments are closed for this post type */
	if ( ( is_page() ) || ( is_single() ) )
		return;

	comment_form();

}

add_filter( 'comment_form_defaults', 'basejump_comment_form_args' );
/**
 * Filters the default comment form arguments, used by <code>comment_form()</code>
 *
 * copied from Genesis
 *
 * @global string $user_identity Display name of the user
 * @global integer $id Post ID to generate the form for
 *
 * @param array $defaults Comment form defaults
 *
 * @return array Filterable array
 */
function basejump_comment_form_args( $defaults ) {

	global $user_identity, $id;

	$commenter = wp_get_current_commenter();
	$req       = get_option( 'require_name_email' );
	$aria_req  = ( $req ? ' aria-required="true"' : '' );

	$author = '<p class="comment-form-author">' .
	          '<input id="author" name="author" type="text" value="' . esc_attr( $commenter['comment_author'] ) . '" size="30" tabindex="1"' . $aria_req . ' />' .
	          '<label for="author">' . __( 'Name', 'basejump' ) . '</label> ' .
	          ( $req ? '<span class="required">*</span>' : '' ) .
	          '</p><!-- #form-section-author .form-section -->';

	$email = '<p class="comment-form-email">' .
	         '<input id="email" name="email" type="text" value="' . esc_attr( $commenter['comment_author_email'] ) . '" size="30" tabindex="2"' . $aria_req . ' />' .
	         '<label for="email">' . __( 'Email', 'basejump' ) . '</label> ' .
	         ( $req ? '<span class="required">*</span>' : '' ) .
	         '</p><!-- #form-section-email .form-section -->';

	$url = '<p class="comment-form-url">' .
	       '<input id="url" name="url" type="text" value="' . esc_attr( $commenter['comment_author_url'] ) . '" size="30" tabindex="3" />' .
	       '<label for="url">' . __( 'Website', 'basejump' ) . '</label>' .
	       '</p><!-- #form-section-url .form-section -->';

	$comment_field = '<p class="comment-form-comment">' .
	                 '<textarea id="comment" name="comment" cols="45" rows="8" tabindex="4" aria-required="true"></textarea>' .
	                 '</p><!-- #form-section-comment .form-section -->';

	$args = array(
		'fields'               => array(
			'author' => $author,
			'email'  => $email,
			'url'    => $url,
		),
		'comment_field'        => $comment_field,
		'title_reply'          => __( 'Have your say...', 'basejump' ),
		'comment_notes_before' => '',
		'comment_notes_after'  => '',
	);

	/** Merge $args with $defaults */
	$args = wp_parse_args( $args, $defaults );

	/** Return filterable array of $args, along with other optional variables */
	return apply_filters( 'basejump_comment_form_args', $args, $user_identity, $id, $commenter, $req, $aria_req );

}

// add tip to contextual help menu, source: https://gist.github.com/2276418 
// Hat tip to Paul de Wouters for all his input: https://plus.google.com/u/0/108543145122756748887/posts/FcBfQ72BpeL
add_action( 'load-appearance_page_custom-background', 'basejump_add_help_tab_to_custom_backgroud_page' );
function basejump_add_help_tab_to_custom_backgroud_page()
	{
		get_current_screen()->add_help_tab( array(
			'id'      => 'custom-background-tip', // This should be unique for the screen.
			'title'   => __('Useful Tip', 'basejump'),
			'content' => basejump_help_tab_content('so-cb-help')
		// Use 'callback' instead of 'content' for a function callback that renders the tab content.
	) );
	}

function basejump_help_tab_content($tab = 'so-cb-help') {
	if($tab == 'so-cb-help') {
		ob_start(); ?>
			<h3><?php _e('Great Resource for Custom Background Images', 'basejump'); ?></h3>
			<p><?php _e('A great source for custom background images is <a href="http://subtlepatterns.com/" title="Subtle Patterns" target="_new" rel="nofollow">Subtle Patterns</a>.', 'basejump'); ?></p>
		<?php
		return ob_get_clean();
	}
}
